app/models/measure.rb

```diff
@@ -553,6 +555,19 @@ def meursing_measures
     @meursing_measures ||= MeursingMeasureFinderService.new(self, meursing_additional_code_id).call
   end
 
+  def sort_key
+    end_date_key = values.key?(:effective_end_date) ? :effective_end_date : :validity_end_date
+
+    [
+      geographical_area_id,
+      measure_type_id,
+      additional_code_type_id,
+      additional_code_id,
+      ordernumber,
+      values[end_date_key] || FAR_FUTURE_END_DATE,
```

Comment:
Lol. I'm waiting for 3023 like :skull_and_crossbones:

Reply:
Haha - yeh, it was a bit of a hacky solution to ruby unable to compare a date to a nil - SQL just orders it at the end so I picked a (very) far off date :)